src/main/java/redis/clients/jedis/CacheJedis.java

```diff
@@ -0,0 +1,265 @@
+        clientTracking();
+        pubSubThread();
+    }
+
+    public CacheJedis(JedisShardInfo shardInfo) {
+        super(shardInfo);
+        clientTracking();
+        pubSubThread();
+    }
+
+    public CacheJedis(URI uri) {
+        super(uri);
+        clientTracking();
+        pubSubThread();
+    }
+
+    public CacheJedis(URI uri, SSLSocketFactory sslSocketFactory, SSLParameters sslParameters, HostnameVerifier hostnameVerifier) {
+        super(uri, sslSocketFactory, sslParameters, hostnameVerifier);
+        clientTracking();
+        pubSubThread();
+    }
+
+    public CacheJedis(URI uri, int timeout) {
+        super(uri, timeout);
+        clientTracking();
+        pubSubThread();
+    }
+
+    public CacheJedis(URI uri, int timeout, SSLSocketFactory sslSocketFactory, SSLParameters sslParameters, HostnameVerifier hostnameVerifier) {
+        super(uri, timeout, sslSocketFactory, sslParameters, hostnameVerifier);
+        clientTracking();
+        pubSubThread();
+    }
+
+    public CacheJedis(URI uri, int connectionTimeout, int soTimeout) {
+        super(uri, connectionTimeout, soTimeout);
+        clientTracking();
+        pubSubThread();
+    }
+
+    public CacheJedis(URI uri, int connectionTimeout, int soTimeout, SSLSocketFactory sslSocketFactory, SSLParameters sslParameters, HostnameVerifier hostnameVerifier) {
+        super(uri, connectionTimeout, soTimeout, sslSocketFactory, sslParameters, hostnameVerifier);
+        clientTracking();
+        pubSubThread();
+    }
+
+    public CacheJedis(URI uri, int connectionTimeout, int soTimeout, int infiniteSoTimeout, SSLSocketFactory sslSocketFactory, SSLParameters sslParameters, HostnameVerifier hostnameVerifier) {
+        super(uri, connectionTimeout, soTimeout, infiniteSoTimeout, sslSocketFactory, sslParameters, hostnameVerifier);
+        clientTracking();
+        pubSubThread();
+    }
+
+    public CacheJedis(URI uri, JedisClientConfig config) {
+        super(uri, config);
+        clientTracking();
+        pubSubThread();
+    }
+
+    public CacheJedis(JedisSocketFactory jedisSocketFactory) {
+        super(jedisSocketFactory);
+        clientTracking();
+        pubSubThread();
+    }
+
+    public CacheJedis(JedisSocketFactory jedisSocketFactory, JedisClientConfig clientConfig) {
+        super(jedisSocketFactory, clientConfig);
+        clientTracking();
+        pubSubThread();
+    }
+    // TODO : If we want to have cache writes , we should have NOLOOP with Broadcasting Mode
+    public void clientTracking() {
+        this.sendCommand(CLIENT , SafeEncoder.encode("TRACKING") , SafeEncoder.encode("on") ,
+                SafeEncoder.encode("REDIRECT") , SafeEncoder.encode(clientId)); //CLIENT TRACKING REDIRECT ON clientId
+    }
+    //Subscribing the pubSub channel in separate thread so that it is non blocking
+    public void pubSubThread()
+    {
+        Runnable runnable = () -> {
+            jedisSub.subscribe(jedisPubSub, "__redis__:invalidate");
```

Comment:
`__redis__:invalidate` make this a private static final constant 

no direct usage of string in code